File: August 2022/August_15.java
class Solution {                                     
    public int romanToInt(String s) {
        Map<Character, Integer> map = new HashMap<>();
        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);
        int num = map.get(s.charAt(s.length() - 1));
        for(int i=s.length() - 2; i >= 0; i--){
            int x = map.get(s.charAt(i));
            if(x >= map.get(s.charAt(i + 1)))
                num += x;
            else
                num -= x;
        }
        return num;
    }
}

File: August 2022/August_15.cpp
class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char, int> mp;
        mp['I'] = 1;
        mp['V'] = 5;
        mp['X'] = 10;
        mp['L'] = 50;
        mp['C'] = 100;
        mp['D'] = 500;
        mp['M'] = 1000;
        int num = mp[s[s.length() - 1]];
        for(int i = s.length() - 2; i >= 0; i--){
            int x = mp[s[i]];
            if(x >= mp[s[i + 1]])
                num += x;
            else
                num -= x;
        }
        return num;
    }
};
